x-pack/plugins/task_manager/server/lib/bulk_update_error.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export class BulkUpdateError extends Error {
+  private _statusCode: number;
+  private _type: string;
+
+  constructor({
+    statusCode,
+    message,
+    type,
+  }: {
+    statusCode: number;
+    message?: string;
```

Comment:
I don't think this should be optioonal.  No idea what `Error()` does :-)

I think the potential `undefined` comes from the task_store usage, accessing `item.update.error.reason`, which is optional.  Let's use a message like `bulk update failed with unknown reason`

Reply:
Done